File: 0226-invert-binary-tree/0226-invert-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        
        def DFS(node):
            # DFS postorder
            # L
            if node.left is not None:
                DFS(node.left)
            # R
            if node.right is not None:
                DFS(node.right)
            # D
            SWAP(node) # swap要最後做   

        def SWAP(node):
            temp = node.left
            node.left = node.right
            node.right = temp

        if root is not None:
            DFS(root)
            
        return root

File: 0226-invert-binary-tree/0226-invert-binary-tree.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */
struct TreeNode* invertTree(struct TreeNode* root) {
    if(root == NULL){
        return NULL;
    }

    // D
    struct TreeNode* temp = root->left;
    root->left = root->right;
    root->right = temp;
    // L
    invertTree(root->left);
    // R
    invertTree(root->right);

    return root;
}